File: src/token/token.zig
const TokenType = enum {
    Illegal,
    Eof,
    Ident,
    Int,
    Assign,
    Plus,
    Comma,
    Semicolon,
    Lparen,
    Rparen,
    Lbrace,
    Rbrace,
    Function,
    Let,
};

pub const Token = struct {
    kind: TokenType,
    literal: []const u8,
};

File: src/token/token.rs
pub enum TokenType {
    Illegal,
    Eof,
    Ident(String),
    Int(Sring),
    Assign(String),
    Plus(String),
    Comma(String),
    Semicolon(String),
    Lparen(String),
    Rparen(String),
    Lbrace(String),
    Rbrace(String),
    Function(String),
    Let(String),
}

pub struct Token {
    pub kind: TokenType,
    pub literal: String,
}
